bundler/lib/bundler.rb

```diff
@@ -497,7 +497,11 @@ def mkdir_p(path)
 
     def which(executable)
       executable_path = find_executable(executable)
-      return executable_path if executable_path
+      if executable_path
+        # add relative path in case of `exec`
+        return "./" + executable_path unless Pathname(executable_path).absolute?
```

Comment:
I'd use `File.absolute_path?(executable_path)` here since we don't use `Pathname` anywhere else in `Bundler.which` implementation.